### PR TITLE
fix(components): [TreeSelect] checkbox interaction

### PR DESCRIPTION
**Pull request by yujinpan**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

Fix:

- https://github.com/element-plus/element-plus/issues/7917
- multiple selection will expand many nodes
屏幕录制2022-06-05 17 47 48 (1)

Docs: 
- update checkbox examples and tips

### CODE REVIEW

**Comment by chenxch** [comment]
Sorry, I'm not very familiar with this component and I need some time to understand.

**Comment by yujinpan** [comment]
> Sorry, I'm not very familiar with this component and I need some time to understand.

Thank you!

File: packages/components/tree-select/src/tree.ts [hunk 1]
```diff
@@ -113,14 +121,11 @@ export const useTree = (
...
-      if (
-        (props.checkStrictly
-          ? props.showCheckbox
-            ? props.checkOnClickNode
-            : props.checkStrictly
-          : props.checkStrictly) ||
-        node.isLeaf
-      ) {
+      // `onCheck` is trigger when `checkOnClickNode`
+      if (props.showCheckbox && props.checkOnClickNode) return
```
Comment (chenxch) [inline packages/components/tree-select/src/tree.ts:125]: Can I understand that the judgment logic of this part has been changed?

Comment (yujinpan) [reply packages/components/tree-select/src/tree.ts:125]: No, checks are blocked from triggering twice here. In order to solve the problem of checking anomalies with only [show checkbox], see the new unit test. Their final result is the same, just as your previous unit tests have not been changed.

Comment (chenxch) [reply packages/components/tree-select/src/tree.ts:125]: All right

**Review by chenxch — APPROVED** [review]

**Pull request merged**